action/action_deserializer.go

```diff
@@ -6,15 +6,93 @@
 
 package action
 
-import "github.com/iotexproject/iotex-proto/golang/iotextypes"
+import (
+	"github.com/iotexproject/go-pkgs/crypto"
+	"github.com/iotexproject/iotex-core/config"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"github.com/pkg/errors"
+)
 
 // Deserializer de-serializes an action
 type Deserializer struct {
+	withChainID bool
 }
 
 // ActionToSealedEnvelope converts protobuf to SealedEnvelope
 func (ad *Deserializer) ActionToSealedEnvelope(pbAct *iotextypes.Action) (SealedEnvelope, error) {
-	var selp SealedEnvelope
-	err := selp.LoadProto(pbAct)
+	var (
+		selp SealedEnvelope
+		err  error
+	)
+	if ad.withChainID {
+		err = selp.LoadProtoWithChainID(pbAct)
+	} else {
+		err = selp.LoadProto(pbAct)
+	}
 	return selp, err
 }
+
+// WithChainID sets whether or not to use chainID
+func (ad *Deserializer) WithChainID(with bool) *Deserializer {
+	ad.withChainID = with
+	return ad
+}
+
+// LoadProtoWithChainID use chainID while loading protobuf
+func (sealed *SealedEnvelope) LoadProtoWithChainID(pbAct *iotextypes.Action) error {
```

Comment:
why is this func public?

Reply:
1. the func is public since it will be used in `block_deserializer.go`
2. after the HF, this func will be removed/reverted, anyway, move it to `sealed_envelope.go`
3. yes, i commented above, in the future when removing global `evmNetworkID`, it will be like 
```
Deserializer.SetEvmNetworkID(id).ActionToSealedEnvelope()
```